jarviscli/plugins/qr_generator.py

```diff
@@ -0,0 +1,168 @@
+        """
+        checks if the URL the user gave is in valid form(can be reached).
+        The user can type 'end' to exit in case his URL is continuously rejected.
+        The method covers the basic exceptions created by invalid URLs.
+        """
+
+        while True:
+            try:
+                response = jarvis.input(prompt,Fore.CYAN)
+                # the user wants to exit
+                if response == "end":
+                    break
+                # tries to get website from URL 
+                website = requests.get(response)
+                break
+            # list of possible exceptions from URLs with invalid form
+            except requests.exceptions.MissingSchema:
+                jarvis.say("\nSorry, your input URL is not valid.",Fore.RED)
+                jarvis.say("Try something in this form 'https://github.com/.../...'")
+            # handles the exception where the user misses part of the 'http' part of the URL
+            except requests.exceptions.InvalidSchema:
+                jarvis.say("\nSorry, your input URL is not valid.",Fore.RED)
+                jarvis.say("Try something in this form 'https://github.com/.../...'")
+            except requests.exceptions.ConnectionError:
+                jarvis.say("\nSorry, your input URL is not valid.",Fore.RED)
+                jarvis.say("Try something in this form 'https://github.com/.../...'")
+            except requests.exceptions.InvalidURL:
+                jarvis.say("\nSorry, your input URL is not valid.",Fore.RED)
+                jarvis.say("Try something in this form 'https://github.com/.../...'")
+                
+        return response
+    
+
+    def _get_path(self, prompt, jarvis):
+        """
+        checks if the path the user gave is valid(directory exists on the computer).
+        The user can type 'end' to exit in case his path is continuously rejected.
+        """
+
+        while True:
+            path = jarvis.input(prompt,Fore.CYAN)
+            # the user wants to exit
+            if path == "end":
+                response = path
+                break
+            isFile = os.path.isdir(path)
+            # directory exists
+            if isFile:
+                response = path
+                break
+            jarvis.say("\nSorry, the path that you specified couldn't be found",Fore.RED)
+            jarvis.say("Try something in this form 'C:\..\..' if you are using Windows for example.")
+            
+        return response
+    
+
+    def _generate_qr(self, jarvis, query):
+        try:
+            # send request
+            jarvis.spinner_start('Creating QR ')
+            response = requests.get(query)
+            # where the png file will be stored
+            location = self.path + self.filename + ".png"
+            file = open(location, "wb")
+            file.write(response.content)
+            jarvis.spinner_stop()
+            file.close
+        except BaseException:
```

Comment:
try-excepting any Exception shadows what kind of exception accrued and makes stuff harder to debug. For example I had Exception:
``[Errno 13] Permission denied: '/tmp\\x.png'``

which was definitely no API error ;). I would recomend:
* Min ``except BaseException as e`` and then ``print(e)``
* Better: Check what exception can actually be thrown by api errors and only catch those